src/Cache/Signature.php

```diff
@@ -73,4 +73,19 @@ public function equals(SignatureInterface $signature)
 
         return true;
     }
+
+    private static function utf8Encode(array $data)
+    {
+        if (!function_exists('utf8_encode') || !function_exists('mb_detect_encoding')) {
```

Comment:
what about requiring `symfony/polyfill-mbstring` and remove both checks, having cache works all the time...